src/transformers/tokenization_utils.py

```diff
@@ -2086,35 +2276,27 @@ def batch_encode_plus(
                             "index {} is of type {}".format(i, type(sample))
                         )
 
-                    # Convert to tuple for convenience
-                    if isinstance(sample, list):
-                        sample = (sample,)
+                    # Test if we have a pair of sentences by checking the depth of nesting
+                    is_pair = bool(len(sample) > 0 and isinstance(sample[0], (list, tuple)))
 
-                    encodings_text = Encoding.merge(self._tokenizer.encode_batch(sample[0], False), True)
+                    # Take care of the first sequence - we multi-thread over the words
+                    encodings_text = EncodingFast.merge(
+                        self._tokenizer.encode_batch(sample[0] if is_pair else sample, add_special_tokens=False),
+                        growing_offsets=True,
+                    )
 
-                    # Check if we have pairs
-                    if len(sample) == 2:
```

Comment:
Checking pairs vs single on `list` vs `tuple` plus the length == 2 is too fragile.
Switching this to test on the depth of nesting (pair inputs have one additional level of nesting versus single inputs)